src/core/agent/services/connectionService.ts

```diff
@@ -299,21 +304,22 @@ class ConnectionService extends AgentService {
   async getConnectionHistoryById(
     connectionId: string
   ): Promise<ConnectionHistoryItem[]> {
-    let histories: ConnectionHistoryItem[] = [];
-    const credentialRecords =
-      await this.credentialStorage.getCredentialMetadataByConnectionId(
+    const linkedIpexMessages =
+      await this.ipexMessageStorage.getIpexMessageMetadataByConnectionId(
         connectionId
       );
-    histories = histories.concat(
-      credentialRecords.map((record) => {
+
+    const requestMessages = linkedIpexMessages
+      .sort((a, b) => a.createdAt.getTime() - b.createdAt.getTime())
```

Comment:
We can just do `b.createdAt.getTime() - a.createdAt.getTime()` and remove return it here, instead of calling `.reverse()`